File: app/src/main/java/team/triplog/adapter/TripLogAdapter.java
package team.triplog.adapter;

import android.content.Context;
import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.shapes.OvalShape;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import team.triplog.R;
import team.triplog.entity.TripLog;

public class TripLogAdapter extends RecyclerView.Adapter<TripLogAdapter.ViewHolder> {
    private Context context;
    private ArrayList<TripLog> tripLogs;

    public TripLogAdapter(Context context, ArrayList<TripLog> tripLogs) {
        this.tripLogs = tripLogs;
        this.context = context;
    }

    @Override
    public int getItemCount() {
        return tripLogs.size();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater
                        .from(context)
                        .inflate(R.layout.item_trip_log, parent, false)
        );
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        TripLog tripLog = tripLogs.get(position);

        if (position == 0) {
            holder.view_trip_log_image.setBackgroundResource(R.drawable.button_plus);
            holder.view_trip_log_delee.setVisibility(View.GONE);
            return;
        }

        holder.textView.setText(tripLog.title);
        holder.view_trip_log_image.setBackgroundResource(tripLog.drawbleId);

    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView textView;
        View view_trip_log_image;
        View view_trip_log_delee;

        ViewHolder(View view) {
            super(view);

            textView = view.findViewById(R.id.text);
            view_trip_log_image = view.findViewById(R.id.view_trip_log_image);
            view_trip_log_delee = view.findViewById(R.id.view_trip_log_delee);
        }
    }
}

File: app/src/main/java/team/triplog/presentation/adapter/TripLogAdapter.kt
package team.triplog.presentation.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_trip_check.view.*
import kotlinx.android.synthetic.main.item_trip_log.view.*
import team.triplog.R
import team.triplog.domain.model.TripLog
import java.util.ArrayList

class TripLogAdapter (
    private val context: Context,
    private val tripLogs: ArrayList<TripLog>
) : RecyclerView.Adapter<TripLogAdapter.ViewHolder>() {

    override fun getItemCount(): Int = tripLogs.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TripLogAdapter.ViewHolder {
        return TripLogAdapter.ViewHolder(
            LayoutInflater
                .from(parent.context)
                .inflate(R.layout.item_trip_log, parent, false)
        )
    }

    override fun onBindViewHolder(holder: TripLogAdapter.ViewHolder, position: Int) {
        val item = tripLogs[position]

        if (position == 0) {
            holder.view_trip_log_image.setBackgroundResource(R.drawable.button_plus);
            holder.view_trip_log_delee.visibility = View.GONE
            return;
        }

        holder.textView.text = item.title
        holder.view_trip_log_image.setBackgroundResource(item.drawbleId)
    }




    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var textView = view.text
        var view_trip_log_image = view.view_trip_log_image
        var view_trip_log_delee = view.view_trip_log_delee
    }

}